examples/libs/ton/basic-tx/main.ts

```diff
@@ -0,0 +1,133 @@
+import { DfnsWallet } from '@dfns/lib-ton'
+import { DfnsApiClient } from '@dfns/sdk'
+import { AsymmetricKeySigner } from '@dfns/sdk-keysigner'
+import { getHttpEndpoint } from '@orbs-network/ton-access'
+import { Address, SendMode, TonClient, WalletContractV4, beginCell, internal, storeMessageRelaxed } from '@ton/ton'
+
+
+import * as dotenv from 'dotenv'
+
+dotenv.config()
+
+const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(hex.replace(/^0x/, ''), 'hex')
+}
+
+const initDfnsWallet = async (walletId: string) => {
+  const signer = new AsymmetricKeySigner({
+    credId: process.env.DFNS_CRED_ID!,
+    privateKey: process.env.DFNS_PRIVATE_KEY!,
+  })
+
+  const dfnsClient = new DfnsApiClient({
+    appId: process.env.DFNS_APP_ID!,
+    authToken: process.env.DFNS_AUTH_TOKEN!,
+    baseUrl: process.env.DFNS_API_URL!,
+    signer,
+  })
+
+  return DfnsWallet.init({
+    walletId: walletId,
+    dfnsClient,
+  })
+}
+
+async function main() {
+  const senderWallet = await initDfnsWallet(process.env.TON_WALLET_ID!)
+  console.log('ton sender address: %s', senderWallet.address)
+
+  // use orbs to connect TON nodes
+  const endpoint = await getHttpEndpoint({
+    network: 'testnet'
+  })
+
+  const client = new TonClient({ endpoint })
+
+  // instance of your wallet (replace with the right version)
+  const tonWallet = WalletContractV4.create({
+    workchain: 0,
+    publicKey: hexToBuffer(senderWallet.publicKey),
+  })
+
+    const opened = client.open(tonWallet)
+    const seqno = await opened.getSeqno()
+
+    const signingMessageBuilder = beginCell().storeUint(tonWallet.walletId, 32)
+    if (seqno === 0) {
+      for (let i = 0; i < 32; i++) {
+        signingMessageBuilder.storeBit(1)
+      }
+    } else {
+      signingMessageBuilder.storeUint(Math.floor(Date.now() / 1e3) + 60, 32)
+    }
+
+    signingMessageBuilder.storeUint(seqno, 32)
+    signingMessageBuilder.storeUint(0, 8) // Simple order
+
+    signingMessageBuilder.storeUint(SendMode.PAY_GAS_SEPARATELY, 8)
+
+    const message = internal({
+      value: '1',
+      to: 'EQBfYLuQwjbBd-LAZ6eNC26XmVVxEl86MQPKG981hdTSicL_',
+      body: 'Example transfer body',
+    })
+    signingMessageBuilder.storeRef(beginCell().store(storeMessageRelaxed(message)))
+
+    const cell = signingMessageBuilder.endCell()
+
+    const signedCell = await senderWallet.sign(cell)
+    
+    await opened.send(signedCell)
+    console.log('native transfer cell broadcasted')
+
+    // unfortunately, we can't have the txHash without pulling the chain
+    const txHash = await retry(() => tryGetTxByBoc(client, tonWallet.address, signedCell.hash().toString('hex'), 10), {
```

Comment:
instead of calling the blockchain directly, can't we just poll DFNS Api to retrieve the hash?